ci-operator/step-registry/azure/provision/resourcegroup/azure-provision-resourcegroup-ref.yaml

```diff
@@ -9,5 +9,16 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
+  env:
```

Comment:
I am thinking if it is worth to write a separate step for hypershift case.

Reply:
I can create a separate step, although this avoids duplicating code. if you feel strongly that this should be a separate step, please lmk.

Reply:
Sounds not a big problem to avoid duplicating codes, cc @jinyunma for double confirm.

Reply:
looks good to me.
But if more pre-configuration steps need to be updated for hypershift testing(mount hypershift azure credentials and change to use this creds in the script), I guess it is deserved to create new cluster profile for the test, then we can reuse those steps directly without any change.

Reply:
@Patryk-Stefanski 

Had a discussion with @jinyunma. 

Ideally, we could add a key named `osServicePrincipal.json` to the `cluster-secrets-hypershift` secret in the vault. The content will be the Azure credentials used in the job, specifically the ones stored in the `hypershift-ci-jobs-azurecreds` secret. This approach allows us to obtain the credentials from `${CLUSTER_PROFILE_DIR}/osServicePrincipal.json`, thereby avoiding the need to mount the `hypershift-ci-jobs-azurecreds` secret to all steps involved in hypershift Azure jobs.

However, if the above solution is not feasible for now for any reason, this PR can be merged as is. The secret mount issue can be addressed later.

Reply:
Yes, ideally `osServicePrincipal.json` can be gotten dynamically which is bound with the `cluster_profile`, no need to mount manually the credential.

Reply:
yeah, I don't see any issues with the suggested approach but with that being said we use the same approach of mounting cred secrets with our aws jobs. I'd like to run this by @sjenning , but he's on PTO till Monday. In the meantime could we merge this without the change and if Seth approves of the new approach then we can address it in a separate pr next week?